plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java

```diff
@@ -0,0 +1,404 @@
+
+      gradleSourceSet.setGradleVersion(project.getGradle().getGradleVersion());
+      gradleSourceSet.setProjectName(project.getName());
+      String projectPath = project.getPath();
+      gradleSourceSet.setProjectPath(projectPath);
+      gradleSourceSet.setProjectDir(project.getProjectDir());
+      gradleSourceSet.setRootDir(project.getRootDir());
+
+      String variantName = (String) variant.getClass().getMethod("getName").invoke(variant);
+      gradleSourceSet.setSourceSetName(variantName);
+
+      // classes task equivalent in android (assembleRelease)
+      gradleSourceSet.setClassesTaskName(
+          SourceSetUtils.getFullTaskName(projectPath, "assemble" + capitalize(variantName))
+      );
+
+      gradleSourceSet.setCleanTaskName(SourceSetUtils.getFullTaskName(projectPath, "clean"));
+
+      // compile task in android (compileReleaseJavaWithJavac)
+      HashSet<String> tasks = new HashSet<>();
+      String compileTaskName = "compile" + capitalize(variantName) + "JavaWithJavac";
+      tasks.add(SourceSetUtils.getFullTaskName(projectPath, compileTaskName));
+      gradleSourceSet.setTaskNames(tasks);
+
+      String projectName = SourceSetUtils.stripPathPrefix(projectPath);
+      if (projectName.isEmpty()) {
+        projectName = project.getName();
+      }
+      String displayName = projectName + " [" + variantName + ']';
+      gradleSourceSet.setDisplayName(displayName);
+
+      // module dependencies
+      Set<GradleModuleDependency> moduleDependencies =
+          AndroidDependencyCollector.getModuleDependencies(project, variant);
+      // add Android SDK
+      Object androidComponents = getAndroidComponentExtension(project);
+      if (androidComponents != null) {
+        Object sdkComponents = getProperty(androidComponents, "sdkComponents");
+        Object bootClasspath =
+            ((Provider<?>) getProperty(sdkComponents, "bootclasspathProvider")).get();
+        try {
+          List<RegularFile> bootClasspathFiles =
+              (List<RegularFile>) bootClasspath.getClass().getMethod("get").invoke(bootClasspath);
+          List<File> sdkClasspath =
+              bootClasspathFiles.stream().map(RegularFile::getAsFile).collect(Collectors.toList());
+          for (File file : sdkClasspath) {
+            moduleDependencies.add(mockModuleDependency(file.toURI()));
+          }
+        } catch (IllegalStateException | InvocationTargetException e) {
+          // failed to retrieve android sdk classpath
+          // do nothing
+        }
+      }
+      // add R.jar file
+      String taskName = "process" + capitalize(variantName) + "Resources";
+      Task processResourcesTask = project.getTasks().findByName(taskName);
+      if (processResourcesTask != null) {
+        Object output = processResourcesTask.getClass()
+            .getMethod("getRClassOutputJar").invoke(processResourcesTask);
+        RegularFile file = (RegularFile) output.getClass()
+            .getMethod("get").invoke(output);
```

Comment:
How about wrap the reflections to a utility method?

Reply:
Addressed in https://github.com/microsoft/build-server-for-gradle/pull/173/commits/e4cc0c244f988508817e6e77796159e9bd0a61d9